File: assemblyscript-algorithms/src/algorithms/basic/sum.ts
export function sum_uint64_asm(array: StaticArray<u64>): u64 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_int64_asm(array: StaticArray<i64>): i64 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_float64_asm(array: StaticArray<f64>): f64 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_float32_asm(array: StaticArray<f32>): f32 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_uint32_asm(array: StaticArray<u32>): u32 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_int32_asm(array: StaticArray<i32>): i32 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_uint16_asm(array: StaticArray<u16>): u16 {
    array.length
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_int16_asm(array: StaticArray<i16>): i16 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_uint8_asm(array: StaticArray<u8>): u8 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}
export function sum_int8_asm(array: StaticArray<i8>): i8 {
  let sum = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    sum += item;
  }
  return sum;
}

File: c-algorithms/src/basic/sum/sum.cpp
#include "basic/sum/sum.h";

u64 sum_u64_c(Array1d_u64* array) {
    u64 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
i64 sum_i64_c(Array1d_i64* array) {
    i64 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
f64 sum_f64_c(Array1d_f64* array) {
    f64 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
f32 sum_f32_c(Array1d_f32* array) {
    f32 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
u32 sum_u32_c(Array1d_u32* array) {
    u32 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
i32 sum_i32_c(Array1d_i32* array) {
    i32 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
u16 sum_u16_c(Array1d_u16* array) {
    u16 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
i16 sum_i16_c(Array1d_i16* array) {
    i16 sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
u8 sum_u8_c(Array1d_u8* array) {
    u8  sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
i8 sum_i8_c(Array1d_i8* array) {
    i8  sum = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        sum += item;
    }
    return sum;
}
